crates/bevy_ui/src/update.rs

```diff
@@ -37,35 +37,46 @@ fn update_clipping(
     entity: Entity,
     clip: Option<Rect>,
 ) {
-    let (node, global_transform, style, calculated_clip) = node_query.get_mut(entity).unwrap();
-    // Update this node's CalculatedClip component
-    match (clip, calculated_clip) {
+    let (node, global_transform, style, old_clip) = node_query.get_mut(entity).unwrap();
+
+    // Update current node's CalculatedClip component
+    match (old_clip, clip) {
         (None, None) => {}
-        (None, Some(_)) => {
+        (Some(_), None) => {
             commands.entity(entity).remove::<CalculatedClip>();
         }
-        (Some(clip), None) => {
+        (None, Some(clip)) => {
             commands.entity(entity).insert(CalculatedClip { clip });
         }
-        (Some(clip), Some(mut old_clip)) => {
+        (Some(mut old_clip), Some(clip)) => {
             if old_clip.clip != clip {
                 *old_clip = CalculatedClip { clip };
             }
         }
     }
```

Comment:
I was going to suggest to use `set_if_neq` but then I checked and we never actually query for changes in `CalculatedClip` so the neq gate is unnecessary: 

```suggestion
        (Some(mut old_clip), Some(clip)) =>  *old_clip = CalculatedClip { clip },
    }
```